tortoise_interpreter.py

```diff
@@ -20,41 +19,59 @@ def sanitize_program(program):
 
 
 def navigate_command_functions(COMMANDS, program):
-    # Comment
-    return [f'{i}. {COMMANDS[command[0]](command)}' for i, command in enumerate(program, 1)]
+    command_to_parse = []
+    pen_state = ''
+    for i, command in enumerate(program, 1):
+        new_pen_state = get_pen_state(command, pen_state)
```

Comment:
The aim was to move the knowledge about when to alter the state into the handler functions. This hasn't been achieved. The handler function should either alter the state it's given, or return a new state.